### PR TITLE
Updated example to work with React 16.6 & Quagga 0.12

### PR DESCRIPTION
**Pull request by macmorning**

### CODE REVIEW

**Comment by macmorning** [comment]
Includes:
* moving from old class create to the more standard "extends React.Component" method
* Quagga instance configuration:
     "facing" => "facingMode"
     readers : ["ean_reader"]

**Comment by henryjw** [comment]
Tried this out on my phone and it definitely fixes the problems with the current example code not working on the browser.

File: src/components/Scanner.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
-import React from 'react';
+import React from "react";
```
Comment (ericblade) [inline src/components/Scanner.js]: Please keep usage of double-quotes and single-quotes consistent

File: src/components/Scanner.js [hunk 2]
```diff
@@ -18,16 +21,16 @@ export default React.createClass({
...
             },
             locator: {
                 patchSize: "medium",
                 halfSample: true
             },
-            numOfWorkers: 2,
+            numOfWorkers: 1,
             decoder: {
-                readers : [ "code_128_reader"]
+                readers : ["ean_reader"]
```
Comment (ericblade) [inline src/components/Scanner.js]: perhaps if we're going to change the default reader, we should probably include a few more common ones? personally i use upc and ean for scanning products and isbn codes, seems there's a good chunk of people who would probably be server by having default be upc, ean, code_128 ?

File: src/index.js [hunk 3]
```diff
@@ -4,5 +4,5 @@ import App from './components/App';
 
 ReactDOM.render(
     <App />,
-    document.getElementById('app')
+    document.getElementById('app'),
```
Comment (ericblade) [inline src/index.js]: i'd have to dig through the code, but i'm pretty sure that the repo follows the "use commas at end" convention for things like this, i'd prefer if this non-functional comma change were not included

**Comment by ericblade** [comment]
@macmorning thanks for the change, and i apologize that it's taken somewhere north of forever for me to look at it.  It does look good, other than a couple of admittedly nitpicky comments wrt style, but I'd prefer not to break the conventions used throughout the source when possible.  

In regards to the change in the webpack config, have you tested that that doesn't break anything?  I'm concerned because I don't know webpack at all, so I'm not good for reviewing that piece.

**Comment by ericblade** [comment]
i just realized i don't have merge access to this repo, only the main one.  :-S

**Comment by macmorning** [comment]
Hey @ericblade no problem; honestly I cloned the repo with very selfish intentions first, I didn't think I would send a PR.
I'll try and fix all this.
I'll double check with webpack but I'm not a big user either.

**Comment by ericblade** [comment]
just a couple lines, really :-) 

I'm thinking now that I understand how npm works, I might get to forking the quagga repos and running maintenance myself.

**Comment by macmorning** [comment]
Corrected and tested, webpack builds without errors.

**Comment by ericblade** [comment]
that's great. :-D thank you very much @macmorning ... unfortunately, i don't have merge access to this repo, but the changes are appreciated!

**Comment by andrralv** [comment]
Why isn't this merged into master already?

**Comment by ericblade** [comment]
@andrralv unfortunately, serratus has become quite busy with things outside of github, it seems.  While I and at least one other have merge privileges for the main repo, we don't have for this one.  I keep talking about making a fork for maintenance until such a time as serratus comes back, or someone else steps up, but i've also been really really busy outside of github, also.

**Comment by ericblade** [comment]
New fork updated for react 16.9 + hooks 

https://github.com/ericblade/quagga2-react-example

**Comment by macmorning** [comment]
Please see this instead :
https://github.com/ericblade/quagga2-react-example

**Pull request closed without merging**